src/main/java/eu/dissco/orchestration/backend/service/MachineAnnotationServiceService.java

```diff
@@ -61,60 +72,112 @@ public class MachineAnnotationServiceService {
   private final ObjectMapper mapper;
   private final MachineAnnotationServiceProperties properties;
   private final KubernetesProperties kubernetesProperties;
+  private final FdoProperties fdoProperties;
 
   private static String getName(String pid) {
-    return pid.substring(pid.indexOf('/') + 1).toLowerCase();
+    return pid.substring(pid.lastIndexOf('/') + 1).toLowerCase();
+  }
+
+  private static SchemaContactPoint__1 buildContactPoint(SchemaContactPoint schemaContactPoint) {
+    return new SchemaContactPoint__1()
+        .withSchemaDescription(schemaContactPoint.getSchemaDescription())
+        .withSchemaEmail(schemaContactPoint.getSchemaEmail())
+        .withSchemaUrl(schemaContactPoint.getSchemaUrl())
+        .withSchemaTelephone(schemaContactPoint.getSchemaTelephone());
   }
 
-  public JsonApiWrapper createMachineAnnotationService(MachineAnnotationService mas, String userId,
+  public JsonApiWrapper createMachineAnnotationService(MachineAnnotationServiceRequest masRequest,
+      String userId,
       String path) throws ProcessingFailedException {
-    var requestBody = fdoRecordService.buildCreateRequest(mas, ObjectType.MAS);
+    var requestBody = fdoRecordService.buildCreateRequest(masRequest, ObjectType.MAS);
     try {
       var handle = handleComponent.postHandle(requestBody);
-      setDefaultMas(mas, handle);
-      var masRecord = new MachineAnnotationServiceRecord(handle, 1, Instant.now(), userId, mas,
-          null);
-      repository.createMachineAnnotationService(masRecord);
-      createDeployment(masRecord);
-      publishCreateEvent(handle, masRecord);
-      return wrapSingleResponse(handle, masRecord, path);
+      setDefaultMas(masRequest, handle);
+      var mas = buildMachineAnnotationService(masRequest, 1, userId, handle, Instant.now());
+      repository.createMachineAnnotationService(mas);
+      createDeployment(mas);
+      publishCreateEvent(mas);
+      return wrapSingleResponse(mas, path);
     } catch (PidCreationException | PidAuthenticationException e) {
       throw new ProcessingFailedException(e.getMessage(), e);
     }
 
   }
 
-  private void setDefaultMas(MachineAnnotationService mas, String handle) {
-    if (mas.getTopicName() == null) {
-      mas.setTopicName(getName(handle));
+  private MachineAnnotationService buildMachineAnnotationService(
+      MachineAnnotationServiceRequest mas, int version, String userId, String handle,
+      Instant created) {
+    var id = HANDLE_PROXY + handle;
+    return new MachineAnnotationService()
+        .withId(id)
+        .withOdsID(id)
+        .withType("ods:MachineAnnotationService")
```

Comment:
we have the ObjectType enum already. Maybe we can modify it to include the type name and use it in these kinds of situations?

Reply:
Done